test/common/http/conn_manager_impl_test.cc

```diff
@@ -3359,6 +3359,59 @@ TEST_F(HttpConnectionManagerImplTest, FilterClearRouteCache) {
   conn_manager_->onData(fake_input, false);
 }
 
+// Dummy PerConnectionSharedPtr creator-function; this creator-function can be as robust
+// and state-preserving as the inheriting interface desires.
+PerConnectionObjectSharedPtr test_creation_function(MutableHttpConnection&) {
+  return std::make_shared<PerConnectionObject>();
+}
+
+TEST_F(HttpConnectionManagerImplTest, MutableHttpConnection) {
+  setup(false, "", true, true);
+
+  MutableHttpConnection mutable_connection;
+  Http::StreamFilterCallbacks::PerConnectionObjectCreator creator_function = test_creation_function;
+  // Types of per connection objects
+  const std::string test_str_obj1 = "read_buffer_handle";
```

Comment:
nit: I don't think having these as different variables helps test readability. Can you just inline this below?